src/dom/Draggable.js

```diff
@@ -202,18 +202,19 @@ export const Draggable = Evented.extend({
 		DomUtil.enableImageDrag();
 		DomUtil.enableTextSelection();
 
-		if (this._moved && this._moving) {
+		const fireDragend = this._moved && this._moving;
 
+		this._moving = false;
+		Draggable._dragging = false;
+
+		if (fireDragend) {

```

Comment:
@mourner @jonkoops @IvanSanchez do you see any problems with setting the flags to `false` before fireing the event?

Reply:
It seems like the only way this would matter would be if someone was looking at `_moving` or `_dragging` during the handling of a `dragend` event.  We're not doing it internally to Leaflet, but it's possible that someone who depends on us might.  Maybe mark this as a breaking API change in case someone depends on the state of `_moving` or `_dragging` in an event triggered by `dragend`?